samtranslator/model/sam_resources.py

```diff
@@ -2144,26 +2147,76 @@ class SamGraphQLApi(SamResourceMacro):
         resources: List[Resource] = [appsync_api, appsync_schema]
 
+        if cloudwatch_role:
+            resources.append(cloudwatch_role)
+
         return resources
 
-    def _construct_appsync_api(self) -> GraphQLApi:
+    def _construct_appsync_api_resources(self) -> Tuple[GraphQLApi, Optional[IAMRole]]:
         api = GraphQLApi(logical_id=self.logical_id, depends_on=self.depends_on, attributes=self.resource_attributes)
 
         api.AuthenticationType = sam_expect(self.Auth.get("Type"), self.logical_id, "Auth.Type").to_be_a_string()
         api.Name = self.Name if self.Name else self.logical_id
         api.XrayEnabled = self.XrayEnabled
 
+        cloudwatch_role = None  # conditionally generated, if left as None then its not added to resources
+
+        # Logging has 3 possible types: dict, bool, and None.
+        # GraphQLApi will not include logging if and only if the user explicity sets Logging as false boolean.
+        # It will for every other value (including true boolean which is essentially same as None).
+        if type(self.Logging) != bool or self.Logging is True:
+            api.LogConfig = self._parse_logging_properties()
+
+            # We create a CloudWatch role for the user if the Logging property is not a dictionary
+            # that contains the "CloudWatchLogsRoleArn" property.
+            if type(self.Logging) != dict or "CloudWatchLogsRoleArn" not in self.Logging:
+                cloudwatch_role = self._construct_cloudwatch_role()
+
         if self.Tags:
             api.Tags = get_tag_list(self.Tags)
 
-        return api
+        return api, cloudwatch_role
+
+    def _parse_logging_properties(self) -> LogConfigType:
+        log_config: LogConfigType = {}
+
+        if type(self.Logging) == dict and "CloudWatchLogsRoleArn" in self.Logging:
+            log_config["CloudWatchLogsRoleArn"] = self.Logging["CloudWatchLogsRoleArn"]
+        else:
+            log_config["CloudWatchLogsRoleArn"] = fnGetAtt(f"{self.logical_id}CloudWatchRole", "Arn")
+
+        if type(self.Logging) == dict and "ExcludeVerboseContent" in self.Logging:
+            log_config["ExcludeVerboseContent"] = self.Logging["ExcludeVerboseContent"]
+
+        if type(self.Logging) == dict and "FieldLogLevel" in self.Logging:
+            log_config["FieldLogLevel"] = self.Logging["FieldLogLevel"]
+        else:
+            log_config["FieldLogLevel"] = "ALL"
+
+        return log_config
+
+    def _construct_cloudwatch_role(self) -> IAMRole:
+        role = IAMRole(
+            logical_id=f"{self.logical_id}CloudWatchRole",
+            depends_on=self.depends_on,
+            attributes=self.resource_attributes,
+        )
+        role.AssumeRolePolicyDocument = IAMRolePolicies.construct_assume_role_policy_for_service_principal(
+            "appsync.amazonaws.com"
+        )
+        role.ManagedPolicyArns = [
+            f"arn:{ArnGenerator.get_partition_name()}:iam::aws:policy/service-role/AWSAppSyncPushToCloudWatchLogs"
```

Comment:
We shouldn't have to fetch the partition ourselves as it causes `cfn-lint` (see https://github.com/aws/serverless-application-model/pull/2550 and https://github.com/aws/serverless-application-model/issues/1590) to complain about hardcoded partitions. You should use `${AWS::Partition}` instead.

Reply:
Good to know, I'll change that